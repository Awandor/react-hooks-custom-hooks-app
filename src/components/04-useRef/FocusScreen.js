import { useRef } from "react";

export const FocusScreen = () => {

    const inputRef = useRef( /* initialValue */ );

    const handleClick = ( e )=>{

        // document.querySelector( 'input' ).focus();

        // document.querySelector( 'input' ).select(); // Pone el foco al seleccionar lo escrito

        // Con el inputRef ya puedo acceder al elemento

        inputRef.current.select();

    };
    
    return (
        <div className="container">
            <h1>Focus Screen</h1>
            <hr/>

            <div className="form-group">
                <label>Nombre</label>
                <input ref={inputRef} type="text" className="form-control" />
            </div>
            <div className="form-group">
                <button className="btn btn-primary" onClick={handleClick}>Focus</button>
            </div>
        </div>
    );

};
