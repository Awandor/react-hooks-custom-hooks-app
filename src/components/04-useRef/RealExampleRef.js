import { useState } from "react";
import { MultipleCustomHooks } from "../03-examples/MultipleCustomHooks"

export const RealExampleRef = () => {

    const [ show, setShow ] = useState( false );

    return (
        <div className="container">
            <h1>Real Example useRef</h1>
            <hr />
            <div className="row">
                <div className="col">
                    {show && <MultipleCustomHooks />}
                </div>
            </div>
            <div className="row mt-4">
                <div className="col">
                    <button type="button" className="btn btn-primary" onClick={() => {

                        setShow( !show );

                    }}>Toggle show</button>
                </div>
            </div>
        </div>
    );

};
