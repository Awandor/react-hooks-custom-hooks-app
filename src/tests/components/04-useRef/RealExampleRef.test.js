import { shallow } from 'enzyme';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';


describe( 'Prueba de <RealExampleRef />', () => {

    test( 'debe mostrarse correctmente', () => {

        const wrapper = shallow( <RealExampleRef /> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find( 'MultipleCustomHooks' ).exists() ).toBeFalsy();
        
    } );

    test( 'debe mostrar el componente <MultipleCustomHooks />', () => {

        const wrapper = shallow( <RealExampleRef /> );

        // Con simulate simulamos eventos y como segundo argumento toma un objeto donde eviamos datos

        wrapper.find( 'button' ).at( 0 ).simulate( 'click', {} );

        const textoParrafo = wrapper.find( '.col' ).at( 0 ).text().trim();

        // console.log( textoParrafo );

        expect( textoParrafo ).toBe( '<MultipleCustomHooks />' );
        expect( wrapper.find( 'MultipleCustomHooks' ).exists() ).toBeTruthy();
        
    } );

    test( 'debe ocultar el componente <MultipleCustomHooks />', () => {

        const wrapper = shallow( <RealExampleRef /> );

        // Al simular el click de nuevo deber√≠a ocultar el componente

        wrapper.find( 'button' ).at( 0 ).simulate( 'click', {} );
        wrapper.find( 'button' ).at( 0 ).simulate( 'click', {} );

        const textoParrafo = wrapper.find( '.col' ).at( 0 ).text().trim();

        console.log( textoParrafo );

        expect( textoParrafo ).toBe( '' );
        expect( wrapper.find( 'MultipleCustomHooks' ).exists() ).toBeFalsy();
        
    } );
    
} );
