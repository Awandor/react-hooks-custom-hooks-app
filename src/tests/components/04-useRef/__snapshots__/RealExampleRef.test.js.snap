// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba de <RealExampleRef /> debe mostrarse correctmente 1`] = `
<div
  className="container"
>
  <h1>
    Real Example useRef
  </h1>
  <hr />
  <div
    className="row"
  >
    <div
      className="col"
    />
  </div>
  <div
    className="row mt-4"
  >
    <div
      className="col"
    >
      <button
        className="btn btn-primary"
        onClick={[Function]}
        type="button"
      >
        Toggle show
      </button>
    </div>
  </div>
</div>
`;
